src/modm/math/matrix_impl.hpp

```diff
@@ -30,6 +30,13 @@ modm::Matrix<T, ROWS, COLUMNS>::Matrix(const T *data)
 	}
 }
 
+// ----------------------------------------------------------------------------
+template<typename T, uint8_t ROWS, uint8_t COLUMNS>
+modm::Matrix<T, ROWS, COLUMNS>::Matrix(const std::initializer_list<T> data)
+{
+	std::copy(data.begin(), data.end(), element);
+}
```

Comment:
The way `initializer_list` is designed there is no way to check the count of elements at compile time because the function argument is not a constant expression. That code is quite dangerous because it could initialize more elements and corrupt the stack or initialize less and leave values undefined.

An better alternative is using variadic templates. That way you can enforce the correct number of values.
```suggestion
template<typename T, uint8_t ROWS, uint8_t COLUMNS>
template<typename... U>
modm::Matrix<T, ROWS, COLUMNS>::Matrix(U... data)
	requires (std::is_convertible_v<U, T> && ...)
	: element{T(data)...}
{
	static_assert(sizeof...(data) == ROWS * COLUMNS, "Invalid element count");
}

```

Reply:
Wizardry :magic_wand:
Very nice :-D thanks!